### PR TITLE
Add delete API support in operator

### PR DESCRIPTION
**Pull request by AmaliMatharaarachchi**
## Purpose

- [x] Add operator controllers to ability to handle API CR deletes.
- [x] Update xds caches upon an API delete 
- [x] Remove old project zip processing implementation as they are no longer valid for APK
- [x] Improve logs 

Fixes https://github.com/wso2/apk/issues/281

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: adapter/internal/operator/controllers/dp/httproute_controller.go [hunk 1]
```diff
@@ -78,15 +79,19 @@ func NewHttpRouteController(mgr manager.Manager, ods *synchronizer.OperatorDataS
 func (r *HttpRouteReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	var httpRoute gwapiv1b1.HTTPRoute
 	if err := r.client.Get(ctx, req.NamespacedName, &httpRoute); err != nil {
-		loggers.LoggerAPKOperator.ErrorC(logging.ErrorDetails{
-			Message:   fmt.Sprintf("httpRoute related to reconcile request with key : %v not found", req.NamespacedName),
-			Severity:  logging.TRIVIAL,
-			ErrorCode: 2611,
-		})
-		// TODO: Handle HttpRoute delete event.
+		apiRef, found := r.ods.HTTPRouteToAPIRefs[req.NamespacedName]
```
[the remark this answers is not in the document]
Comment (AmaliMatharaarachchi) [reply adapter/internal/operator/controllers/dp/httproute_controller.go]: Fixed with https://github.com/wso2/apk/pull/293/commits/fc1555e1ab3df455df0d3bc23758b0dceeedd49f. Could you please take a look again

File: adapter/internal/discovery/xds/server_utils.go [hunk 2]
```diff
@@ -33,7 +34,7 @@ func getEnvironmentsToBeDeleted(existingEnvs, deleteEnvs []string) (toBeDel []st
 	}
 	// otherwise delete env if it wished to
 	for _, existingEnv := range existingEnvs {
-		if arrayContains(deleteEnvs, existingEnv) {
+		if stringutils.ArrayContains(deleteEnvs, existingEnv) {
```
Comment (Amila-Rukshan) [inline adapter/internal/discovery/xds/server_utils.go:37]: existing `stringutils.StringInSlice` and new utils`stringutils.ArrayContains` are same. shall we use the existing one?

File: adapter/pkg/utils/stringutils/string_utils.go [hunk 3]
```diff
@@ -57,3 +57,13 @@ func StringInSlice(a string, list []string) bool {
 	}
 	return false
 }
+
+// ArrayContains returns if the string is in array
+func ArrayContains(stringArray []string, stringElement string) bool {
```
Comment (Amila-Rukshan) [inline adapter/pkg/utils/stringutils/string_utils.go:62]: we can remove this after using `StringInSlice`

File: adapter/internal/operator/controllers/dp/httproute_controller.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (NomadXD) [reply adapter/internal/operator/controllers/dp/httproute_controller.go]: Deletion flow seems okay with the API CR deletion. Also as we discussed it's better to finalize the other inputs like HttpRoute deletions, updates etc.

[the remark this answers is not in the document]
Comment (AmaliMatharaarachchi) [reply adapter/internal/operator/controllers/dp/httproute_controller.go]: yes, finalizing will be addressed by another PR

**Review by Amila-Rukshan — APPROVED** [review]

**Pull request merged**